js-check/2.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * Task 2
+ *
+ */
+
+const user = {
+    firstName: 'John',
+    lastName: 'Doe',
+    rate: 0.3,
+    address: {
+        line1: '15 Macon St',
+        line2: '',
+        city: 'Gotham'
+    },
+    phoneNumbers: [
+        {
+            type: 'MOBILE',
+            number: '(555) 555-1234'
+        },
+        {
+            type: 'LINE',
+            number: '(555) 555-5678'
+        }
+    ]
+};
+
+let checkingPhoneNumber = (objects) => {
+    const MOBILE = 'MOBILE ';
+    const LINE = 'LINE';
+    const VOIP = 'VOIP';
+    const PHONEREG = /^\(([0-9]{3})\)[ ]([0-9]{3})[-]([0-9]{4})$/;
+
+    return objects.every(obj => obj.type === MOBILE || LINE || VOIP && PHONEREG.test(obj.number));
+};
+
+let checkField = (obj) => {
+    if (
```

Comment:
in general, I would rather like to see sequence of checks then one huge `if` statement